src/services/importService.ts

```diff
@@ -61,33 +65,61 @@ export default class ImportService implements IImportService {
             },
             autoSave: true,
         };
+
+        if (originalProject.visitedFrames.every((item) => typeof item === "number")) {
+            // then this is a video project
+            convertedProject.lastVisitedAssetId = "dummyId";
+        }
+
         return convertedProject;
     }
 
     /**
      * Generate assets based on V1 Project frames and regions
      * @param project - V1 Project Content and File Information
      */
-    public async generateAssets(v1Project: IFileInfo, v2Project: IProject): Promise<IAssetMetadata[]> {
+    public async generateAssets(v1Project: IFileInfo, v2Project: IProject, parent?: IAsset): Promise<IAssetMetadata[]> {
         let originalProject: IV1Project;
         const generatedAssetMetadata: IAssetMetadata[] = [];
         let assetState: AssetState;
         const assetService = new AssetService(v2Project);
+        let isVideo = false;
 
         originalProject = JSON.parse(v1Project.content as string);
 
         for (const frameName in originalProject.frames) {
             if (originalProject.frames.hasOwnProperty(frameName)) {
                 const frameRegions = originalProject.frames[frameName];
-                const asset = AssetService.createAssetFromFilePath(
-                    `${v1Project.file.path.replace(/[^\/]*$/, "")}${frameName}`);
-                const populatedMetadata = await assetService.getAssetMetadata(asset).then((metadata) => {
+                let asset: IAsset;
+                // if video:
+                if (originalProject.visitedFrames.every((item) => typeof item === "number") && parent !== null) {
```

Comment:
I recommend splitting this function up into multiple smaller functions to handle the different between images & video.